infrastructure/containers/postgres/automatic-backup/backup_or_restore.py

```diff
@@ -145,17 +212,18 @@ def restore_database(backup_fpath):
 
 
 def main():
+    print("System args:", sys.argv)
     print("Running database-backup process with subcommand", sys.argv[1])
     if sys.argv[1] == "backup":
-        backup_database(make_backup_fpath())
+        backup_database(make_object_name())
     elif sys.argv[1] == "backup-on-interval":
         backup_interval_seconds = parse_timedelta(BACKUP_INTERVAL).seconds
         backup_database_on_interval(seconds=backup_interval_seconds)
     elif sys.argv[1] == "restore-from-most-recent":
-        restore_database(get_most_recent_backup_fpath())
+        restore_database()
     elif sys.argv[1] == "restore-from-backup":
-        backup_fpath = BACKUP_DIR / sys.argv[2]
-        restore_database(backup_fpath)
+        backup_obj_name = sys.argv[2]
+        restore_database(backup_obj_name)
     else:
         print(
             dedent(
```

Comment:
Wow. Seriously, _nice_ work on this file. I wouldn't have felt comfortable having actual rootski users without these changes. It's virtually guaranteed that our database will go down at some point. This is going to give us a really elegant way to recover from that. It also will make deployment way easier. You're the man, @ir3456 !